### ISSUE TITLE
[CBT] Spanish chatbot Interface

### ISSUE THREAD

**Issue by joofio** [Gravitate-Health/mvp-issues#15]
`apply a very tight scoped version of a spanish clinical oriented chat-bot that indicates sections that contains certain info (already this) - link it to the fhir epi and sections of it`

**Comment by amedranogil** [comment]
Create an interface for Chatbot interactions (MVP3.0).
Then Spanish scenario chatbot and others could be integrated.

**Comment by amedranogil** [comment]
Discussing Spanish Scenario's licence to integrate.
initially the integration whould be by calling their service (i.e. FOSPS as a proxy).
Other chatbot implementations may be deployed within FOSPS, but the integration would be the same, via internal call

**Comment by joofio** [comment]
RAG would work with this?
My current vision is:
an LLM that retrieves the epi to answer questions. We can/should use the fhir one so we can link to sections.it will probably need some pre and/or postprocessing to the information.
what do you think?

**Comment by amedranogil** [comment]
Yes, but this would be the implementation of a new chatbot, the Spanish chatbot is already operational.
The Spanish chatbot in order to ensure safety only answers with ePI excerps.

Of course RAG would be the only viable way to make an LLM chatbot safe enough, and developing such chatbot directly compatible with FHIR ePI will certainly push towards the digitisation and the standard.

**Comment by gmej** [comment]
Closing as completed.
